test/WebJobs.Script.Tests/Workers/Rpc/GrpcWorkerChannelTests.cs

```diff
@@ -1355,7 +1355,47 @@ public async Task GetFunctionMetadata_MultipleCalls_ReturnSameTask()
             Assert.Same(functionsTask1, functionsTask2);
         }
 
-        private IEnumerable<FunctionMetadata> GetTestFunctionsList(string runtime, bool addWorkerProperties = false)
+        [Fact]
+        public async Task Log_And_InvocationResult_OrderedCorrectly()
+        {
```

Comment:
Wonder there is a way for us to have more than one invocations and logs being send in overlapping times of them to see they are processed correctly.

Reply:
There's no way for those messages to overlap with the current implementation of how we handle grpc messages. They come through a single-threaded Channel and are processed one-at-a-time. 

This test shows how, even today, that doesn't work b/c we take that single threaded Channel and dispatch to the ThreadPool, introducing a race condition that breaks all ordering.